DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -308,11 +356,22 @@ function IsPackageInstalled
     $installedPackages = Get-ChocoInstalledPackage
 
     if ($pVersion) {
-        Write-Verbose 'Comparing version'
-        $installedPackages = $installedPackages | Where-object { $_.Name -eq $pName -and $_.Version -eq $pVersion}
-    } else {
+        Write-Verbose 'Comparing required version'
+        $installedPackages = $installedPackages | Where-Object { $_.Name -eq $pName -and $_.Version -eq $pVersion}
+    }
+    elseif ($pMinimumVersion) {
+        Write-Verbose 'Comparing minimum version'
+        $comparablePackages = $installedPackages | Where-Object { $_.Name -eq $pName} | ForEach-Object {
+            # to do version comparision we first need to convert the version from string to System.Version
+            # we also need to ignore any pre release version, i.e. anythign after "-"
+            $v = [System.Version](($_.Version -split "-")[0])
+            $_ | Add-Member -MemberType NoteProperty -Name ComparableVersion -Value $v -PassThru
+        }

```

Comment:
This isn't going to work for pre-release versions?

Reply:
good spot, maybe one for discussion. Short answer is no because the way SemVer works doesn't line up with how `[System.Version]` works. For now I have added in a throw to make this clear to users that you cant use MinimumVersion with pre-release. 

However, is this acceptable? Happy to consider another solution. From my point of view this is okay since you probably want to tightly control a pre-release in DSC anyway. If you wanted a specified a pre-release in MinimumVersion you could potentially end up with anything as more pre-release are pushed which isnt very good for immutability.

Worth noting also that out of the box Version doesn't support pre-release versions either until you pass `--pre` via chocoParams.